src/midifunctor.cpp

```diff
@@ -717,6 +717,19 @@ FunctorCode GenerateMIDIFunctor::VisitPedal(const Pedal *pedal)
 {
     if (!pedal->HasDir()) return FUNCTOR_CONTINUE;
 
+    // Check if the pedal applies to the staff filtered
+    if (this->GetFilters()) {
+        Pedal *nonConstPedal = const_cast<Pedal *>(pedal);
+        Measure *measure = vrv_cast<Measure *>(nonConstPedal->GetFirstAncestor(MEASURE));
+        assert(measure);
+        std::vector<Staff *> staffList = nonConstPedal->GetTstampStaves(measure, nonConstPedal);
```

Comment:
If we use pedal without `staff` attribute, then `GetTstampStaves` returns an empty list. This filters out any pedal events. But I guess in this case the pedal should be applied to all staves instead of none.